android/src/amazon/java/com/dooboolab/RNIap/RNIapAmazonModule.java

```diff
@@ -53,41 +53,31 @@
     "PROMISE_QUERY_AVAILABLE_ITEMS";
   public static final String PROMISE_GET_USER_DATA = "PROMISE_GET_USER_DATA";
 
-  private final ReactContext reactContext;
 
-  private LifecycleEventListener lifecycleEventListener;
-  private PurchasingListener purchasingListener = null;
 
   public RNIapAmazonModule(final ReactApplicationContext reactContext) {
     super(reactContext);
-    this.reactContext = reactContext;
 
-    lifecycleEventListener =
+    UiThreadUtil.runOnUiThread(() -> {
```

Comment:
This new implementation follows Amazon's documentation guidelines
https://developer.amazon.com/docs/in-app-purchasing/iap-implement-iap.html#4-implement-getpurchaseupdates-method